### PR TITLE
Typescript support

### PR DESCRIPTION
**Pull request by imeredith**
This is a WIP. Please comment on any improvements that need to be done. 

I doubt this will work with tests as it stands.

### CODE REVIEW

**Comment by sophistifunk** [comment]
I am happy that this is a thing :)

**Comment by kzantow** [comment]
Should probably figure out a way to swap between babel and typescript or maybe just enable each independently baed on presence of `.babelrc` or `tsconfig.json`

**Comment by imeredith** [comment]
@kzantow TSC is only used on ts/tsx files, and babel is used on the rest, so both work at the same time fine

File: internal/bundlegen.js [hunk 1]
```diff
@@ -222,14 +230,20 @@ exports.doJSBundle = function(bundle, applyImports) {
...
         var babelConfig = {};
 
+
         // if no .babelrc was found, configure babel with the default presets and plugins from above
         if (!hasBabelRc) {
             babelConfig.presets = presets;
             babelConfig.plugins = plugins;
         }
 
+        var babelifyConfig = packageJson.babelify || {};
```
Comment (kzantow) [inline internal/bundlegen.js]:
```
if (packageJson.babelify) {
  bundler.transform(babelify.configure(babelifyConfig), babelConfig);
} else {
  bundler.transform(babelify, babelConfig);
}

**Pull request merged**